packages/openapi3/lib/decorators.tsp

```diff
@@ -13,3 +13,28 @@ extern dec oneOf(target: Union | ModelProperty);
  * @param ref External reference(e.g. "../../common.json#/components/schemas/Foo")
  */
 extern dec useRef(target: Model | ModelProperty, ref: valueof string);
+
+/** Metadata to a single tag that is used by operations. */
+model TagMetadata {
+  /** A description of the API. */
+  description?: string;
+
+  /** An external Docs information of the API. */
+  externalDocs?: ExternalDocs;
+}
+
+/** External Docs information. */
+model ExternalDocs {
+  /** Documentation url */
+  url: string;
+
+  /** Optional description */
+  description?: string;
+}
+
+/**
+ * Specify OpenAPI additional information.
+ * @param name tag name
+ * @param additionalTag Additional information
+ */
+extern dec tagMetadata(target: Namespace, name: valueof string, tagMetadata?: TagMetadata);
```

Comment:
```suggestion
extern dec tagMetadata(target: Namespace, name: valueof string, tagMetadata?: valueof TagMetadata);
```

change to this and you can get rid of most of the JS code

Reply:
old decorators do not use that for legacy purpose but any new decorator that just expect a config value should use values and not models as config

Reply:
@timotheeguerin 
if using `TagMetadata`, it cannot add Extension. `x-custom: "string",`
 currenly, the tag metadate should support as following....
```
{
    description: "Pets operations",
    externalDocs: {
      url: "https://example.com",
      description: "More info.",
      `x-custom-externalDocs`: "string",
    },
    `x-custom`: "string",
  }
```

Reply:
Update to extends record unknown

Reply:
this is still not fixed...
the signature MUST looks like this

```
extern dec tagMetadata(target: Namespace, name: valueof string,  tagMetadata?: valueof TagMetadata)
```
your last change did nothing

Reply:
Fixed